src/twisted/plugins/twisted_reactors.py

```diff
@@ -4,59 +4,70 @@
 from __future__ import absolute_import, division
 
 from twisted.application.reactors import Reactor
-from twisted.python.compat import _PY3
+from twisted.python.reflect import requireModule
 
-default = Reactor(
+def _addReactorToAll(shortName, moduleName, description):
+    """
+
+    @param shortName: Short name of reactor
+    @param moduleName: The fully-qualified module name of the reactor
+    @param description: Description of reactor
+    @return: newly created reactor
+    @rtype: L{Reactor} or L{None}
+    """
+    newReactor = None
+    if requireModule(moduleName) is not None:
```

Comment:
With the existing implementation, if you do:

trial --help-reactors

The list that appears:

(1)   Displays reactors which aren't supported.  For example, if I do that on a Mac,
       I see:
          iocp	Win32 IO Completion Ports-based reactor.
          win32    Win32 WaitForMultipleObjects-based reactor.

     No way these Windows reactors can run on a Mac.

       I also see:
    gi  	GObject Introspection integration reactor.
    glib2	GLib2 event-loop integration reactor.
    gtk2	Gtk2 integration reactor.
    gtk3	Gtk3 integration reactor.
   wx  	wxPython integration reactor.

    but I don't have glib2 or gtk2 or gtk3 or wxPython installed.

   I also see:
     epoll	epoll(4)-based reactor.

   but that is Linux-specific and can't run on Mac.


(2)   Doesn't display reactors which are supported.  For example, if I do this on Windows with Python 3, I *don't* see:
           iocp	Win32 IO Completion Ports-based reactor.
           win32   Win32 WaitForMultipleObjects-based reactor.

But those reactors should be there, because they work on Python 3 now.

So the existing implementation lists reactors it shouldn't, and
doesn't list reactors that it should!